src/Client.js

```diff
@@ -276,6 +285,14 @@ Client.prototype.request = function (options, params, callback) {
     // Add all the given parameters to the querystring.
     req = req.query(queryParams);
 
+    if (isFunction(reqCustomizer)) {
+      reqCustomizer(req, params);
+    }
+
+    if (isFunction(params._requestCustomizer)) {
+      params._requestCustomizer(req, params);
```

Comment:
personally I wouldn't choose "middleware" because it has such a specific meaning in very well known nodejs projects - I think that would have the pontential to confuse people (or possibly give them a false initial impression).

I understand your idea, I personally wouldn't want code to turn off the "global middleware" - it defined as global for a reason (e.g. set some extra query param - e.g. a hashed shared-secret) and it has a potentially high WTF factor if a given request can turn it off, also the `params` are passed into the "middleware" (which allows for conditionally skipping customization based 'something').

in `express`, the quintessential "middleware" based thing, middleware is run unconditionally (each function decides internally whether to just call `next()` or actually do something), and they are "stacked" (i.e. run one after the other - this is how I was thinking about it). I think your idea for renaming to "middleware" actually weakens the argument to change to logic regarding how/when the functions are applied.

I can live with the logic change you propose (I'm actually only using the request-specific middleware in my code) and the name-change ... mostly I'm short on time (so if I can avoid doing any more work on this that's a win :-))

If you could live with the PR as it is that would be awesome, if you really insist then I guess I'll have to put in the extra work to get it in a shape that you accept!